src/DashboardService.Business/Board/RemoveBoardCommand.cs

```diff
@@ -1,25 +1,53 @@
 ﻿using LT.DigitalOffice.DashboardService.Business.Board.Interfaces;
 using LT.DigitalOffice.DashboardService.Data.Interfaces;
+using LT.DigitalOffice.Kernel.BrokerSupport.AccessValidatorEngine.Interfaces;
+using LT.DigitalOffice.Kernel.Extensions;
 using LT.DigitalOffice.Kernel.Helpers.Interfaces;
 using LT.DigitalOffice.Kernel.Responses;
+using Microsoft.AspNetCore.Http;
 using System;
+using System.Net;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace LT.DigitalOffice.DashboardService.Business.Board;
 
 public class RemoveBoardCommand : IRemoveBoardCommand
 {
   private readonly IBoardRepository _boardRepository;
+  private readonly IHttpContextAccessor _httpContext;
   private readonly IResponseCreator _responseCreator;
+  private readonly IAccessValidator _accessValidator;
 
-  public RemoveBoardCommand(IBoardRepository boardRepository, IResponseCreator responseCreator)
+  public RemoveBoardCommand(
+    IBoardRepository boardRepository,
+    IHttpContextAccessor contextAccessor,
+    IResponseCreator responseCreator,
+    IAccessValidator accessValidator)
   {
     _boardRepository = boardRepository;
+    _httpContext = contextAccessor;
     _responseCreator = responseCreator;
+    _accessValidator = accessValidator;
   }
 
-  public Task<OperationResultResponse<bool>> ExecuteAsync(Guid boardId)
+  public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid boardId, CancellationToken ct)
   {
-    throw new NotImplementedException();
+    if (!await _accessValidator.IsAdminAsync())
+    {
+      _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.Forbidden);
+    }
+
+    if (!await _boardRepository.RemoveAsync(boardId, _httpContext.HttpContext.GetUserId(), ct))
```

Comment:
userId can be stored in var and used in IsAdminAsync()

Reply:
++, and also add userId in other methods, where use isAdminAsync()